staging/src/k8s.io/apiserver/pkg/server/serve.go

```diff
@@ -121,52 +118,26 @@ func runServer(server *http.Server, network string, stopCh <-chan struct{}) (int
 		return 0, fmt.Errorf("invalid listen address: %q", ln.Addr().String())
 	}
 
-	lock := sync.Mutex{} // to avoid we close an old listener during a listen retry
+	// Stop the server by closing the listener
 	go func() {
 		<-stopCh
-		lock.Lock()
-		defer lock.Unlock()
 		ln.Close()
 	}()
 
 	go func() {
-		defer utilruntime.HandleCrash()
```

Comment:
I've added this back as per @liggitt.  I misread a comment on ``HandleCrash`` that suggested it might no longer be necessary.